agent/engine/docker_container_engine.go

```diff
@@ -848,3 +857,31 @@ func (dg *dockerGoClient) removeImage(imageName string) error {
 	}
 	return client.RemoveImage(imageName)
 }
+
+// LoadImage invokes loads an image from an input stream, with a specified timeout
+func (dg *dockerGoClient) LoadImage(inputStream io.Reader, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	response := make(chan error, 1)
+	go func() {
+		response <- dg.loadImage(docker.LoadImageOptions{
+			InputStream: inputStream,
+			Context:     ctx,
+		})
+	}()
+	select {
+	case resp := <-response:
+		return resp
+	case <-ctx.Done():
```

Comment:
If a timeout happens, how does the caller know when it's safe to close the `io.Reader`?